core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -1024,10 +1039,10 @@ class Partition(val topicPartition: TopicPartition,
                                      maxNumOffsets: Int,
                                      isFromConsumer: Boolean,
                                      fetchOnlyFromLeader: Boolean): Seq[Long] = inReadLock(leaderIsrUpdateLock) {
-    val localReplica = localReplicaWithEpochOrException(Optional.empty(), fetchOnlyFromLeader)
+    val localLog = localLogWithEpochOrException(Optional.empty(), fetchOnlyFromLeader)
     val allOffsets = logManager.getLog(topicPartition) match {
```

Comment:
Seems we can simplify this. Can't we use `localLog.legacyFetchOffsetsBefore(timestamp, maxNumOffsets)`?

Reply:
Done. However this assumes that locaLog is not None.